### PR TITLE
Upgrade to Windows SDK 10.0.22000.0

### PR DESCRIPTION
**Pull request by JunielKatarn**
## Description
Upgrade to Windows SDK 10.0.22000.0

### Type of Change
- New feature (non-breaking change which adds functionality)

### Why
1. Required for #10870.\
  ARM64EC requires Windows SDK version 10.0.22000.0 to correctly handle intrinsics includes.\
  See https://developercommunity.visualstudio.com/t/arm64ec-toolchain-doesnt-work/1460523\
  \
  Windows SDK 10.0.19041.0 does not ship `softintrin.h`, causing compilation errors on the affected platform.
1. Visual Studio 2022 defaults to Windows SDK 10.0.22000.0.\
  If there is no other specific need to use 10.0.19041.0, the upgrade will also reduce the dev dependency size.

### What
Replaced references to Windows SDK version 10.0.19041.0 with version 10.0.22000.0.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10869)

### CODE REVIEW

File: packages/@react-native-windows/cli/src/runWindows/utils/build.ts [hunk 1]
```diff
@@ -24,12 +24,12 @@ export async function buildSolution(
   buildLogDirectory?: string,
   singleproc?: boolean,
 ) {
-  const minVersion = new Version(10, 0, 19041, 0);
+  const minVersion = new Version(10, 0, 22000, 0);
```
Comment (JunielKatarn) [inline packages/@react-native-windows/cli/src/runWindows/utils/build.ts:27]: Should we keep `19041` as a minimum version?

File: vnext/Scripts/rnw-dependencies.ps1 [hunk 2]
```diff
@@ -61,7 +61,7 @@ $vsComponents = @('Microsoft.Component.MSBuild',
     'Microsoft.VisualStudio.Component.VC.Tools.x86.x64',
     'Microsoft.VisualStudio.ComponentGroup.UWP.Support',
     'Microsoft.VisualStudio.ComponentGroup.NativeDesktop.Core',
-    'Microsoft.VisualStudio.Component.Windows10SDK.19041');
+    'Microsoft.VisualStudio.Component.Windows10SDK.22000');
```
Comment (JunielKatarn) [inline vnext/Scripts/rnw-dependencies.ps1:64]: This is failing PR validation.

Do we need to apply an update to the build agent image?

**Review by jonthysell — CHANGES_REQUESTED** [review]
Please hold on this, it's not as easy (or desirable) to just do a find-replace.

**Comment by JunielKatarn** [comment]
> Please hold on this, it's not as easy (or desirable) to just do a find-replace.

Sure thing. Reverting to draft for now.

Is there a suggested sequence to carry this type of upgrade?

**Comment by jonthysell** [comment]
> > Please hold on this, it's not as easy (or desirable) to just do a find-replace.
> 
> Sure thing. Reverting to draft for now.
> 
> Is there a suggested sequence to carry this type of upgrade?

Well, ideally we don't want to upgrade everyone unless there's some clear requirement. Since Desktop needs it, I think it would be easier to simply override the version when building the Desktop and leave UWP to default targeting 19041.

**Comment by JunielKatarn** [comment]
> Well, ideally we don't want to upgrade everyone unless there's some clear requirement. Since Desktop needs it, I think it would be easier to simply override the version when building the Desktop and leave UWP to default targeting 19041.

That is certainly an option.

I figured it would be convenient to have both variants on the same SDK version.
As stated in the PR description, reducing the installation footprint (VS 2022 defaults to 22000 for the UWP workflow) seemed like a good reason.

Given the min SDK version remains unchanged, what's the criteria to upgrade the target SDK version?
If I read you right, do we prefer to never upgrade unless forced to?

**Pull request closed without merging**